src/plan/global.rs

```diff
@@ -542,6 +546,15 @@ impl<VM: VMBinding> BasePlan<VM> {
     }
 }
 
+#[cfg(feature = "immortal_as_nonmoving")]
+pub type NonMovingSpace<VM> = crate::policy::immortalspace::ImmortalSpace<VM>;
+
+#[cfg(not(any(feature = "immortal_as_nonmoving", feature = "marksweep_as_nonmoving")))]
+pub type NonMovingSpace<VM> = crate::policy::immix::ImmixSpace<VM>;
+
+#[cfg(feature = "marksweep_as_nonmoving")]
+pub type NonMovingSpace<VM> = crate::policy::marksweepspace::native_ms::MarkSweepSpace<VM>;
```

Comment:
If you think it is not worth introducing the `trait NonMovingSpace` as I suggested in https://github.com/mmtk/mmtk-core/pull/1308#issuecomment-2862128839, we can still simplify this part of the code with `cfg_if::cfg_if!`.  The `else` branch can save you from having to type `not(any(feature = "immortal_as_nonmoving", feature = "marksweep_as_nonmoving"))`.

Reply:
See my comments in https://github.com/mmtk/mmtk-core/pull/1308#discussion_r2078718672. I would like to introduce a feature `immix_as_nonmoving` as a default feature. That would solve most of the issues here with annoying `cfg` (we also have the same issue with the features like `malloc_jemalloc`, and `eager_sweep`).